code/game/objects/effects/temporary_visuals/temporary_visual.dm

```diff
@@ -0,0 +1,40 @@
+//temporary visual effects
+/obj/effect/temp_visual
+	icon_state = "nothing"
+	anchored = TRUE
+	layer = ABOVE_MOB_LAYER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	unacidable = 1
+	var/duration = 10 //in deciseconds
+	var/randomdir = TRUE
+	var/timerid
```

Comment:
You don't need to handle this manually, its already handled in https://github.com/discordia-space/CEV-Eris/blob/master/code/datums/datum.dm#L21

Reply:
updated